models/inherit.py

```diff
@@ -28,20 +30,45 @@ def current_user_rents(self):
         
     @api.multi
     def get_rent_count(self):
-        data_obj = self.env['bike.rent']
-        print("*****************************************")
-        print(data_obj.partner_id.name)
-        print("*****************************************")
-
         for record in self:
+            record._compute_rent_count = len(record.partner_ids)
+            print(record._compute_rent_count)
+
+class account_invoice(models.Model):
+    _inherit = 'account.invoice'
+    return_date = fields.Date(string='Bike return date')
+    
+    @api.multi
+    def return_date(self):
+        bike_rent_obj = self.env['sale.order']
+        self.return_date = bike_rent_obj.rent_stop
 
-            print("*****************************************")
-#            record._compute_rent_count = data_obj.search_count(['record.active', '=', True])
-            record._compute_rent_count = data_obj.search_count(['record.name', '=', data_obj.partner_id])
-#            list_data = data_obj.search(['record.name', '=', data_obj.partner_id])
-#            print(len(list_data))
-            print("*****************************************")
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+  
+    rent_start = fields.Date(string='Rent start', required=True)
+    rent_stop = fields.Date(string='Rent stop', required=True)
+    rent_duration = fields.Integer(string='Rent duration')
     
-class Chatter(models.Model):
-    _name = 'user.chatter'
-    _inherit = 'mail.thread'
\ No newline at end of file
+    @api.multi
+    @api.depends('rent_duration')
+    def rent_dates(self):
+        for record in self:
+            record.rent_start = datetime.today().strftime('%Y-%m-%d')
+            record.rent_stop = record.rent_start + timedelta(record.rent_duration)
+        
+    @api.multi
+    def action_confirm(self):
+        self.rent_dates()
+        def filter_rests(record):
+            return record.product_id.is_bike and record.product_id.type == 'service'
+                   
+        for record in self.order_line.filtered(filter_rests):
+            self.env['bike.rent'].create({
+                'price': record.price_subtotal,
+                'rent_start': self.rent_start,
```

Comment:
`self` can be a `recordset` of orders, so not only the for loop would fail here, but every read of `self` would too.